### ISSUE TITLE
LLVM ERROR: Bad $ operand number in inline asm string

### ISSUE THREAD

**Issue by Prince213** [ziglang/zig#7653]
I have built zig a9c75a2b48f202d5c55097877499942ed07cc2e8 for i386-windows-gnu and x86_64-windows-gnu using zig-bootstrap on a fresh Ubuntu 20.10 x86_64.
However, when I try to use 32bit zig on a 64bit Windows 10, Zig shows a mysterious message(64bit binary works fine):
```
zig build-exe test.zig
LLVM Emit Object... LLVM ERROR: Bad $ operand number in inline asm string: ' movl %fs:0x18, $1744424263173013504'
```
test.zig:
```zig
pub fn main() void {}
```

**Comment by daurnimator** [comment]
This is related to how we fetch the TEB: https://github.com/ziglang/zig/blob/a9c75a2b48f202d5c55097877499942ed07cc2e8/lib/std/os/windows.zig#L1474

**Comment by Prince213** [comment]
@daurnimator Just found that this issue can be fixed by #7642.
